Add support for ARM lit testing #618

mgrang opened the pull request.

Commit:
Add support for ARM lit testing #618

Commit:
Update feature name

Commit:
Remove the default X86 target and make it generic.

Commit:
Support ARM lit testing based on llvm-lit flag

We pass a flag -DCHECKEDC_TARGET=ARM to llvm-lit in order to enable ARM testing
for lit.

Commit:
Remove cmake variable CHECKEDC_LIT_TARGETS

dtarditi reviewed (commented):
Could you update the README file to briefly describe the support for ARM emulation testing that you are adding to these files?

Commit:
Add descriptions for newly added lit variables

mgrang commented:
> Could you update the README file to briefly describe the support for ARM emulation testing that you are adding to these files?

I have added descriptions for the newly added variables inline. I have also updated Testing.md with the full details on how to enable lit testing for ARM. PR #645

Commit:
Add error checking for new lit ARM variables

Commit:
Fix error message

dtarditi reviewed (approved):
Looks good, thanks!  You might want to clarify that a non-native target could be emulated or an actual device.
